include/swift/ABI/MetadataValues.h

```diff
@@ -14,14 +14,17 @@
 // includes target-independent information which can be usefully shared
 // between them.
 //
+// This header ought not to include any compiler-specific headers (such as
+// those from `swift/AST`, `swift/SIL`, etc.) since doing so may introduce
```

Comment:
Is it worth adding a test to enforce this, with a whitelist of allowed `swift/XXX` paths?

Also, have we checked the other ABI headers for this problem?

Reply:
Agreed, I'm not a huge fan of that myself.